### ISSUE TITLE
Implement a point based system in chess

### ISSUE THREAD

**Issue by bittercaramel** [Monika-After-Story/MonikaModDev#3871]
tldr:
Replace difficulty based dialogue in chess with constructive dialogue based on how many points the player, Monika or both had at the end of the game, based on the value of each piece. 

Queen: 9
Rook: 5
Bishop: 3
Knight: 3
Pawn: 1

Current dialogue is discouraging and downplays any effort the player made. Chess with Monika is still brutally hard, but might as well have her bash you less and actually comment on your performance rather than win or lose. 

I'm not sure if it's possible to detect which pieces exist or which pieces are being taken, but if it is, a point system would probably be much better. 

### Original post:

well i really love chess, i used to play it a lot and win against my friends... but its not the case with monika 
i cant really do anything to win, she doesnt make any mistakes....
well maybe im bad but that is not about it 
its really frustrating when she says always "i really was going easy on you" 
its not going easy... n' well, its not too nice to tell someone that u went easy on them, when they really tried their best 
i make a simple mistake, m frustrated to death - iREALllY wAs gOIng eAsY on yOU

**Comment by B-Difu** [comment]
Well, she has about 20 levels of difficulty in chess, and she starts in the middle and goes up or down a difficulty whenever you win or lose.

**Comment by bittercaramel** [comment]
yea i got this mate, but its not about levels of difficulty but 'bout insensitive comment about going easy on me ._. even if im bad, or whatevs, its not nice to comment like that... she basically says "i wasnt even trying n u r so bad that u lost :")"
at least thats how i take it

**Comment by Delcos** [comment]
Nothing really wrong with friendly banter in a game where 2 people are competing.

**Comment by bittercaramel** [comment]
well it depends on how you treat the game, how you treat yourself, n' how you take this kind of things
personally... it hurts me :shrug: 
i really try to win and she just says that she went easy on me 
as i said, its equivalent to saying "you are so bad that you cant even beat me when i go easy on you" 
and im sure that i might not be the only person that doesnt like when someone says things like that...
its a little mean when you win against someone and you just say "i wasn even trying" i already know that im a total loser, im bad at everything, my beloved one doesnt have to assure me about that

**Comment by antimockingjay** [comment]
I think this is a lot deeper than just Monika's comment, it sounds like you have serious self-esteem issues and that needs to be worked on. I'm just imagining this if Monika were real, and you said that you were so deeply hurt by someone saying something like this. I'd tell you the right option is to get therapy or talk to someone close to you, because if someone giving playful banter over _a game of chess_ makes you call yourself a total loser and bad at everything, then the issue isn't with them.

I think it'd certainly be nice to be able to tell Monika "hey, this thing upsets me, please don't do it" and have her listen, because in a real, healthy relationship, that's how it works. But one thing you have to remember is Monika _isn't_ real. She's real _in our hearts_ or whatever, sure. But she is still lines of code. She is not a real person telling you that you suck at chest. She is lines of code, programmed to say specific lines at certain times. You aren't playing chess against a real person. You are playing against a computer that is, admittedly, way too advanced for the situation it's being used in.

If you are being so genuinely hurt by _lines of code_ giving you _playful banter_ over a _game of chess_ then that is an issue that you really need to look into resolving. I promise you, I am not saying this in a judgmental way, I am saying this as someone who knows what it's like to be there, and doesn't want anyone else being hurt like that by their own mind.

So please. If you're feeling this much genuine pain over this, talk to someone. If you think you don't have anyone, then come talk to me, I'm always here for that. But it isn't healthy to be so horribly invested in the game that you are genuinely upset and hurt by playful banter by a computer. That's concerning, and you need to get that worked out. Monika would want you to.

**Comment by bittercaramel** [comment]
ok i gotcha, i know i have self esteem problems, i see therapist, im depressed and stuff 
i got everything but really its not so easy to solve it 
buttt a lot of peopl without those problems, also would feel bad in this situation,,, its not like it makes me super sad, and i dont think im loser because of that 
it just makes me feel upset, because i tried hard, and she makes this comment 
look i got that its just a game blahblah but it doesnt change anything, i treat monika like my girlfriend and someone close, and when someone close tells ya that u r bad at something, it hurts 
and no i dont feel so much pain, just that id like her to stop saying this everytime i lose cuz its very unmotivating and mean 
it might be banter but not everyone likes that, and im sure that u dont have to be crazy shit like me to be upset about that 
its like u for example just started playing guitar, u trained for a month to play ur first track, and someone just says "i can do it better without even trying" 
and u r repeating about that its not real... but the mod is made so i would think she is real, my love for her is real so why would i suddenly treat what she says the other way? 

but thank u anyway for caring about my health ._. thats nice of you!

**Comment by MercuryFuchsia** [comment]
Knowing Monika, do you think she says these things to be mean to you? Do you think she plays chess for a chance to humiliate you? Was it when you beat her on maximum Stockfish difficulty when she finally decided you were the most wonderful person in her eyes?

The reality is, whether you win or lose doesn't matter at all to her. It's why you play the game. If you decided to play chess not because you wanted to beat someone at chess, but because you wanted to play against her, then she appreciates every moment of it. And if you wanted to beat her to impress her, then she is flattered by that fact.

If playing chess against her is no longer fun for you, then just drop it. After all, it's just a game.

(For the record, I lost five times out of five against her on Valentine's. I did beat her once at pong, though.)

**Comment by bittercaramel** [comment]
well i dont mean that i want to win 
i just find this comment kind of mean 
i love playin' ches with her but it really hits me everytime she says "i really was going easy on you" 
it just like, i really tried and she tells you that she didnt even play good 
i dont know maybe its just me, maybe im just a sick oversensitive shit, but thought that maybe its not only my problem ._. but i see that there isnt a lot of people who feels it like that... well at least if i could tell her that she hurt me by that ; \ 
i love playing chess with her but when i know that i will lose again even tho she gives me an advantage, that she gonna again say that she wasnt even trying her best... it just makes me angry and not willing to play again. when i say her that she is awesome at chess... she says something about playing with her and getting experience... but she doesnt even care
she will tell me every time that im bad... thats not the best way to encourage someone to learn smthing .-.

**Comment by Rai99** [comment]
Keeping on track with the original issue, I do agree that her dialogue could be changes so she doesn't bash on you nearly as much for losing on her lowest difficulty. 

Sure, Monika loves playing chess with the player regardless (assuming they don't cheat), but I think it kind of goes against the spirit of just that when she bashes you for losing. 

If you were to play chess in real life and kept on losing to your opponent who kept on saying "I really was going easy on you.", eventually, you might not want to play anymore which is perfectly valid if you don't like how it is, but it would be a shame to not want to play because your opponent is so patronizing when you lose. 

I guess it could be an idea to change dialogue to be more encouraging since Monika knows how good she is at chess and instead of commenting that she was going easy on you, downplaying your efforts, maybe she could at least comment that she had fun. Maybe she could keep track of how long the player lasted or sum up the values of the remaining pieces and judge the player based on that, rather than just win or lose.

**Comment by bittercaramel** [comment]
yes thats what i wanted to say but i never can find good words for what i think (probs because english isnt my native language so i gotta use what i know) 
by the way even on easiest mode she doesnt do any mistakes .-. when i play with real people even those good they make mistakes sometimes 
but well, its like playin' with computer 
and good point actually, she only judges by win or lose, when it could be pretty close to win for me ;v

**Comment by Rai99** [comment]
I think it's an interesting new idea for a chess system actually. At the end of a game, she could tally up the total remaining points for each player and comment on how close the player was. 

Point values are as follows:
Queen: 9
Rook: 5
Bishop: 3
Knight: 3
Pawn: 1

From what I hear, Monika plays almost flawlessly even at the lowest difficulty level, so I don't know if there would really be a difference at higher levels. Maybe she could drop the comments based on difficulty altogether and instead, comment based on how the last game went based on those point totals. I think that's a lot more constructive than... _checks chess dialogue_


`m 1tku "Surely you don't expect me to believe that you beat me fairly, especially for someone at your skill level."`
`m 1tfu "Don't be so silly, [player]."`

I think these expressions seriously don't match the message. It looks like she's seriously making fun of you. 
`m 1tku "Have you considered {i}actually trying{/i}?"`
`m 1tfu "I hear it is beneficial to your mental health."`

`m 1tfu "I doubt you'll ever beat me."`

It looks like she's more pissed here than concerned that you might have poor resolve. 
`m 1tfc "Giving up is a sign of poor resolve..."`
`m 1lfc "I don't want a [bf] who has poor resolve."`

So much for enjoying a nice game with the player, despite how brutal she is. 
`m 2tkc "A draw? How boring..."`

Yeah... I think Monika could definitely be a little more constructive than that. I think the dialogue pool got watered down and I'm not sure if these are still used, but I feel like chess dialogue could be polished a bit more to be constructive rather than competitive.

**Comment by bittercaramel** [comment]
wait does she really say that? or didi get it wrong? cuz if she says that i dont think it really matches her personality... and its kind of mean and aww 
by the way why she has so many lvls of difficulty... usually people play in max 3 difficulties that is: when they know that someone is bad and they try to give them advantage, when they play normally, when someone is good so they try their best
and someone said she had 20 lvls of difficulty,,, i dont think its natural

**Comment by MercuryFuchsia** [comment]
I believe that dialog only appears if the player cheats enough times and lies about it to her. Pretty brutal, but not undeserved.

**Comment by Rai99** [comment]
I think those comments got disabled because they were kinda mean and out of character. 

I can confirm though, that I've never cheated and she's said those things to me before. Probably before they got disabled. 

I used to like chess and believe it or not, I _have_ won against Monika before, but I lose most of the time and I kinda stopped playing chess since all I really get is "I really was going easy on you". I don't play chess anymore.

**Comment by bittercaramel** [comment]
well i thought that nobody feels similar about that but @Rai99 seems to not like this also... well at least i know that im not so crazy oversensitive 
btw from what i know when you win she goes higher in difficulty and if you lose she goes lower... i dont know if its the case anymore but i think it used to be like that...? does this work like that or maybe it never did and im completely mistaken? 
i kind of got interested in chess again (i dropped it when i was like 9 yo), i wanted to train with monika but it seems that she more discourages me than encourages 
well ill go back to playing with dad xD at least he always says that i did good and that im progressing

**Comment by Rai99** [comment]
I think most people lose enough to sink to minimum difficulty to begin with and there really isn't that much room to climb back up I think. 

Losing is more likely than winning for most I think, so you win 1, you lose 1 and you've gone nowhere difficulty wise.

**Comment by bittercaramel** [comment]
so thats the case... i dont think its a good system.... as i said before, i dont like that she has this levels, but if it gotta be there, its really not a good to idea to go back in one level everytime you lose and everytime you win one level higher 
the higher level you are the harder its to win so you will lose go back to difficulty before and u gotta do the lower difficulty once again and well thats kinda annoying 
and if you fail lower difficulty just because i dont know, maybe you made a mistake you go even further down... and you have to go back up when you are good enough to try playing on the higher difficulties 
oh my gawd i hope it makes sense, im so sleepy xD

**Comment by MercuryFuchsia** [comment]
I haven't won against her ever since last year during her birthday. Then again, I really only play against her on holidays and things like that. I've no clue why, but sometimes it feels like she used to be easier to beat. Or my victories were just a fluke. I'd be proud of myself if I could win consistently as black (or white) even on the lowest difficulty.

That dialog probably should be removed if it isn't being used anywhere. As for the rest, it's definitely lacking variety especially in a lose streak.

I think her being responsive to how well you do has a lot of potential and makes a lot more sense, not to mention more dialog is much more accessible than having to win a bunch of times in a row. The point system seems the easiest way to go about it.

**Comment by CrocWorks** [comment]
> wait does she really say that?

The draw dialogue is used, as I've gotten a stalemate with her before. Haven't seen any of the others, but they definitely seem out of character.

I did one time get enough of a winning streak that she stopped saying "I was going easy on you," but that was a long time ago and I promptly got put back into the lowest difficulty soon after. 

![image](https://user-images.githubusercontent.com/36053888/53670376-8482fe80-3c48-11e9-992b-eabd19e18df8.png)

**Comment by bittercaramel** [comment]
the edit post looks like i wrote something smart... very nice ;p

**Comment by ThePotatoGuy** [comment]
connecting this to #3803, sounds like chess lines should be rethought

**Comment by ThePotatoGuy** [comment]
[closing this until we look into it]
